docs/user-guide/concepts/tools/python-tools.md

```diff
@@ -281,3 +283,99 @@ When using the [`@tool`](../../../api-reference/tools.md#strands.tools.decorator
 1. If you return a string or other simple value, it's wrapped as `{"text": str(result)}`
 2. If you return a dictionary with the proper [`ToolResult`](../../../api-reference/types.md#strands.types.tools.ToolResult) structure, it's used directly
 3. If an exception occurs, it's converted to an error response
+
+## Class-Based Tools
+
+Class-based tools allow you to create tools that maintain state and leverage object-oriented programming patterns. This approach is useful when your tools need to share resources, maintain context between invocations, or follow object-oriented design principles.
+
+### Basic Example
+
+Here's how to create a class-based tool using the [`@tool`](../../../api-reference/tools.md#strands.tools.decorator.tool) decorator:
+
+```python
+from strands import Agent, tool
+
+class Browser:
+    """Browser tool implementation using Playwright."""
+
+    def __init__(self):
+        self._started = False
+        self._sessions = {}
+        # Initialize other class-level resources or state
+
+    @tool
+    def browser(self, url: str, action: str = "navigate") -> str:
```

Comment:
Not required: I might suggest using weather as an example, as that's where we've used elsewhere for both `@tool` and module-based approach.  The customize option might be what city weather is for or what provider to use, or the # of days